File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.java
class Solution {
    public int solution(int[] array) {
        int answer = 0;
        int[] index = new int[1001];

        for(int i = 0; i<array.length; i++) {
			index[array[i]]++; 
		}
        
        int max = 0;
        int max_index = 0;
        
        for(int i = 0; i<index.length; i++) {
			if(index[i] > max) { 
				    max = index[i];
            max_index = i;
			}
		}
        
        int count = 0;
        for(int i = 0; i<index.length; i++) {
            if(index[i] == max) {
                count++;
            }
        }
        answer = max_index;
        
        if(count > 1) {
            answer = -1;
        }
        return answer;
    }
}

File: 프로그래머스/lv0/120812. 최빈값 구하기/최빈값 구하기.js
function solution(array) {
    let answer = 0;
    let index = new Array(1001).fill(0);

    for (let i = 0; i < array.length; i++) {
        index[array[i]]++;
    }

    let max = 0;
    let max_index = 0;

    for (let i = 0; i < index.length; i++) {
        if (index[i] > max) {
            max = index[i];
            max_index = i;
        }
    }

    let count = 0;
    for (let i = 0; i < index.length; i++) {
        if (index[i] === max) {
            count++;
        }
    }

    answer = max_index;

    if (count > 1) {
        answer = -1;
    }

    return answer;
}
